### PR TITLE
Add docs about the handling of SVG parameters

### PR DESCRIPTION
**Pull request by rduivenvoorde**
Information about the use of SVG parameters (as mentioned in http://blog.geoserver.org/2016/10/31/geoserver-2-10-0-released/ in the section 'Improved QGIS style compatibility").

I worked with that and had to find out by trial and error which parts of the url you need to escape.

I hope this text adds this information to the docs.

With me it looks like:

Screenshot-20200818213627-734x1244

For all pull requests:

- [ X ] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [ X ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: doc/en/user/source/styling/sld/extensions/pointsymbols.rst [hunk 1]
```diff
@@ -286,7 +286,75 @@ For example:
...
     </PointSymbolizer>
 
-In this example an SVG graphic is being used, so the size is specified explicitly. 
+In this example an SVG graphic is being used, so the size is specified explicitly.
+
+Parameterized (like QGIS) SVG images
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Just like QGIS, Geoserver can handle SVG images in which parts of the SVG-attributes are 
+named parameters as defined in the `SVG Parameters 1.0 spec <https://www.w3.org/TR/SVGParamPrimer/>`_.
```
Comment (jodygarnett) [inline doc/en/user/source/styling/sld/extensions/pointsymbols.rst:295]: I did not know this was part of the SVG specification cool :)

* `GeoServer`

* I try and use links of the form:
  ```
  `SVG Parameters 1.0 spec <https://www.w3.org/TR/SVGParamPrimer/>`__
  ```
  
  The two under scores are processed differently by sphinx and avoids a conflict if the same URL is referenced twice in the docs.

File: doc/en/user/source/styling/sld/extensions/pointsymbols.rst [hunk 2]
```diff
@@ -286,7 +286,75 @@ For example:
...
+
+   <svg enable-background="new 0 0 580 580" height="580" viewBox="0 0 580 580" width="580" xmlns="http://www.w3.org/2000/svg">
+   <path d="m290.565 67.281l-255.498 442.534-1.087 1.885 511.229.393 2.18.002z" fill="param(fill)" 
+    fill-opacity="param(fill-opacity)" stroke="param(outline)" stroke-opacity="param(outline-opacity)" stroke-width="param(outline-width)"/>
+   </svg>
+
+The 'param'-constructs mean that you can define the parameters: fill, fill-opacity, outline, outline-opacity and outline-width as so called SVG URL references,
+e.g. a refence to this image with red fill would be: ``poi_peak.svg?fill=#FF0000``.
+
+Note: when you edit svg's yourself (e.g. in Inkscape) you should save your svg's as 'simple svg'.
```
Comment (jodygarnett) [inline doc/en/user/source/styling/sld/extensions/pointsymbols.rst:309]: We try to use ` characters (monospace) for named things as they are displayed different, and not to quote things like "simple svg".

Comment (jodygarnett) [reply doc/en/user/source/styling/sld/extensions/pointsymbols.rst:309]: Our [writing guidelines](https://docs.geoserver.org/latest/en/docguide/style.html) try and avoid the use of "we" and "you" and instead provide direct commands.

```
Note: When editing SVG files (e.g. in Inkscape) save using the `simple svg` format.
``

File: doc/en/user/source/styling/sld/extensions/pointsymbols.rst [hunk 3]
```diff
@@ -286,7 +286,75 @@ For example:
...
+
+Now in the SLD the OnlineResource uri without any parameters in the 'href':
```
Comment (jodygarnett) [inline doc/en/user/source/styling/sld/extensions/pointsymbols.rst:311]: Same feedback on direct commands (avoiding "now"):

```
The SLD the OnlineResource uri without any parameters in the 'href':
``

**Comment by rduivenvoorde** [comment]
@jodygarnett please edit to make it in line with geoserver style! Thanks for this.

**Comment by aaime** [comment]
@jodygarnett @rduivenvoorde are we good here, can this be merged?

**Comment by rduivenvoorde** [comment]
For me yes, I think Jody did add the edits here in github isn't it, so no need to merge those?

**Comment by aaime** [comment]
@jodygarnett so is this good to merge?

**Comment by jodygarnett** [comment]
Yes, I tried making the changes directly, easier than explaining.

There does seem to be a build failure... unlikely to be related.

**Comment by aaime** [comment]
Agree it's unrelated. @jodygarnett please merge as you see fit.

**Comment by jodygarnett** [comment]
Thanks for helping out @rduivenvoorde :)

**Pull request merged**